internal/providers/terraform/aws/s3_bucket.go

```diff
@@ -0,0 +1,324 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/schema"
+)
+
+func GetS3BucketRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_s3_bucket",
+		RFunc: NewS3Bucket,
+	}
+}
+
+func NewS3Bucket(d *schema.ResourceData, u *schema.ResourceData) *schema.Resource {
+	return &schema.Resource{
+		Name:           d.Address,
+		SubResources:   s3SubResources(d),
+		CostComponents: s3CostComponents(d),
+	}
+}
+
+func s3CostComponents(d *schema.ResourceData) []*schema.CostComponent {
+	region := d.Get("region").String()
+
+	costComponents := make([]*schema.CostComponent, 0)
+
+	objTagsEnabled := false
+
+	for _, rule := range d.Get("lifecycle_rule").Array() {
+		if len(rule.Get("tags").Map()) > 0 {
+			objTagsEnabled = true
+		}
+	}
+
+	if objTagsEnabled {
+		costComponents = append(costComponents, &schema.CostComponent{
+			Name:           "Object tagging",
+			Unit:           "tags",
+			UnitMultiplier: 10000,
+			ProductFilter: &schema.ProductFilter{
+				VendorName: strPtr("aws"),
+				Region:     strPtr(region),
+				Service:    strPtr("AmazonS3"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "usagetype", ValueRegex: strPtr("/TagStorage-TagHrs/")},
```

Comment:
our units is tags but this mentions tagHrs, seems confusing

Reply:
Yep it's odd - the AWS unit is "Tag-Mo" 🤦